### ISSUE TITLE
Problem about preprocess step

### ISSUE THREAD

**Issue by 626lt** [ManipTrans/ManipTrans#60]
Hi, Thanks for your wonderful work! We try to use ManipTrans on our data. When I run the mano2dexhand.py, the result is quite 
strange.

https://github.com/user-attachments/assets/c4f04f0a-cd16-46d2-b38b-ff8e5e7577e5

1. The dexhand is shaking abnormally, which is quite different with the grab demo.
2. The fingers should grab the object, but it didn't, though the loss is 0.0523 after 2400 iterations.

I am very grateful if you can help me about the reason and tell me which should I check. Thanks for your reply!

**Comment by 626lt** [comment]
Besides, I conducted reinforcement learning training, and the results seem relatively normal, but only for a very short part of the operations. I wonder if there are any hyperparameters that limit the frame rate. Thanks for your reply!

https://github.com/user-attachments/assets/25440912-0d32-4807-a7b7-4da1d3ee4ffe

**Comment by KailinLi** [comment]
Hi, thanks for your interest in ManipTrans.

From your video, the RL training only produces a very short segment of motion. My guess is that this is caused by an unsatisfactory result in the first-step preprocessing (retargeting). For objects with handles, the collision behavior shown in your video is very typical — it is actually collision, rather than abnormal “shaking”.

You can refer to [#57](https://github.com/ManipTrans/ManipTrans/issues/57#issuecomment-3637120116)￼; this should be a common issue in such cases.

**Comment by 626lt** [comment]
Thanks for your patient reply!  I have question about the suggestion in https://github.com/ManipTrans/ManipTrans/issues/57#issuecomment-3637120116

1. Initialize the fitting based on the hand–object interaction location, rather than the object’s geometric center. This is crucial for objects with handles.

What is the meaning of  initialization of the fitting? It seems that object's geometric center is not mentioned in the mano2dexhand.py 

2. Slightly adjust the dexhand position along the surface normal of the interaction region on the object mesh. This can effectively avoid severe finger–object penetration at the early stage of optimization.

I can roughly understand this point, but I have no idea about how to implement it so that different object pose can use it.

**Comment by KailinLi** [comment]
1.	Here, “initialization of the fitting” means the initial values of the optimization variables. In an optimization problem—especially for interaction cases—the hand’s initial pose/position is very important. In our code, this corresponds to the initialization around here: https://github.com/ManipTrans/ManipTrans/blob/a3d08cfe3c3a5868a7f057533bcaf759c5af4705/main/dataset/mano2dexhand.py#L327.
Regarding the “object’s geometric center”: you’re right that it is not explicitly specified in mano2dexhand.py. However, the origin of the object’s canonical coordinate system is typically defined near its geometric center, so the object translation is also usually around that region.
2.	The object surface normal is defined in the object canonical frame. So for different object poses, you can simply rotate the normal into the world frame by applying the object rotation:
\mathbf{n}_{world} = \mathbf{R}_{obj}\,\mathbf{n}_{canon}.
Then you can offset the dexhand along \mathbf{n}_{world} accordingly.

**Comment by KailinLi** [comment]
Closing this issue due to inactivity. Feel free to reopen if needed. Thank you!
